### PR TITLE
增加基于硬件耗时搜索功能及模型硬件耗时评估工具

### PR DESCRIPTION
**Pull request by lijiancheng0614**
1. 更新搜索策略支持延时评估器约束
    - 底层实现增加约束代码 https://github.com/PaddlePaddle/Paddle/blob/develop/python/paddle/fluid/contrib/slim/nas/light_nas_strategy.py
    - 接口暴露增加约束参数 https://github.com/PaddlePaddle/models/blob/develop/PaddleSlim/light_nas/compress.yaml
2. 增加延时评估器生成工具
    - 增加延时评估器生成工具https://github.com/PaddlePaddle/models/tree/develop/PaddleSlim/light_nas/get_latency_lookup_table.py
    - 支持不同的预测框架，如 Paddle Mobile,  用户自定义预测库
3. 使用文档
    - 原理介绍 https://github.com/PaddlePaddle/models/blob/develop/PaddleSlim/docs/tutorial.md
    - 使用说明 https://github.com/PaddlePaddle/models/blob/develop/PaddleSlim/docs/usage.md
    - 使用示例 https://github.com/PaddlePaddle/models/blob/develop/PaddleSlim/docs/demo.md

### CODE REVIEW

File: PaddleSlim/docs/usage.md [hunk 1]
```diff
@@ -606,20 +618,103 @@ controllers:
...
+    - pool_type (int) - 池化类型，其中 1 表示 pooling_max，2 表示 pooling_average_include_padding，3 表示 pooling_average_exclude_padding。
+    - flag_bias (int) - 是否有 bias（0：无，1：有）。
+    - flag_global_pooling (int) - 是否为全局池化（0：不是，1：是）。
+    - flag_relu (int) - 是否有 relu（0：无，1：有）。
+    - n_in (int) - 输入 Tensor 的批尺寸 (batch size)。
+    - c_in (int) - 输入 Tensor 的通道 (channel) 数。
+    - h_in (int) - 输入 Tensor 的特征高度。
+    - w_in (int) - 输入 Tensor 的特征宽度。
+    - c_out (int) - 输出 Tensor 的通道 (channel) 数。
+    - groups (int) - 卷积二维层（Conv2D Layer）的组数。
```
Comment (baiyfbupt) [inline PaddleSlim/docs/usage.md:665]: groups=>group? 与上文 `conv flag_bias flag_relu n_in c_in h_in w_in c_out group kernel padding stride dilation latency`统一

**Comment by baiyfbupt** [comment]
Please check `PaddleNLP/language_representations_kit/ERNIE` modification in this PR

**Comment by lijiancheng0614** [comment]
> Please check `PaddleNLP/language_representations_kit/ERNIE` modification in this PR

This PR has not changed `PaddleNLP/language_representations_kit/ERNIE`, and it can be automatically merged into develop branch which will not affect `ERNIE`.

**Review by baiyfbupt — APPROVED** [review]
LGTM

File: PaddleSlim/docs/usage.md (hunk 1 above)
Comment (lijiancheng0614) [reply PaddleSlim/docs/usage.md:665]: ok, fixed.

**Review by wanghaoshuang — APPROVED** [review]

**Pull request merged**